test/core/slice/slice_hash_table_test.cc

```diff
@@ -113,128 +114,113 @@ static void test_slice_hash_table() {
   const size_t num_entries_d = GPR_ARRAY_SIZE(test_entries_d);
-  grpc_slice_hash_table* table_d =
+  RefCountedPtr<TestHashTable> table_d =
       create_table_from_entries(test_entries_d, num_entries_d, value_cmp_fn);
-  GPR_ASSERT(grpc_slice_hash_table_cmp(table_a, table_d) < 0);
-  GPR_ASSERT(grpc_slice_hash_table_cmp(table_d, table_a) > 0);
+  GPR_ASSERT(TestHashTable::Cmp(*table_a, *table_d) < 0);
+  GPR_ASSERT(TestHashTable::Cmp(*table_d, *table_a) > 0);
 
   // Same values but different "equals" functions.
-  const test_entry test_entries_e[] = {
+  const TestEntry test_entries_e[] = {
       {"key_0", "value_0"}, {"key_1", "value_1"}, {"key_2", "value_2"}};
   const size_t num_entries_e = GPR_ARRAY_SIZE(test_entries_e);
-  grpc_slice_hash_table* table_e =
+  RefCountedPtr<TestHashTable> table_e =
       create_table_from_entries(test_entries_e, num_entries_e, value_cmp_fn);
-  const test_entry test_entries_f[] = {
+  const TestEntry test_entries_f[] = {
       {"key_0", "value_0"}, {"key_1", "value_1"}, {"key_2", "value_2"}};
   const size_t num_entries_f = GPR_ARRAY_SIZE(test_entries_f);
-  grpc_slice_hash_table* table_f =
+  RefCountedPtr<TestHashTable> table_f =
       create_table_from_entries(test_entries_f, num_entries_f, pointer_cmp_fn);
-  GPR_ASSERT(grpc_slice_hash_table_cmp(table_e, table_f) != 0);
+  GPR_ASSERT(TestHashTable::Cmp(*table_e, *table_f) != 0);
 
   // Same (empty) key, different values.
-  const test_entry test_entries_g[] = {{"", "value_0"}};
+  const TestEntry test_entries_g[] = {{"", "value_0"}};
   const size_t num_entries_g = GPR_ARRAY_SIZE(test_entries_g);
-  grpc_slice_hash_table* table_g =
+  RefCountedPtr<TestHashTable> table_g =
       create_table_from_entries(test_entries_g, num_entries_g, value_cmp_fn);
-  const test_entry test_entries_h[] = {{"", "value_1"}};
+  const TestEntry test_entries_h[] = {{"", "value_1"}};
   const size_t num_entries_h = GPR_ARRAY_SIZE(test_entries_h);
-  grpc_slice_hash_table* table_h =
+  RefCountedPtr<TestHashTable> table_h =
       create_table_from_entries(test_entries_h, num_entries_h, pointer_cmp_fn);
-  GPR_ASSERT(grpc_slice_hash_table_cmp(table_g, table_h) != 0);
-
-  grpc_core::ExecCtx exec_ctx;
-  grpc_slice_hash_table_unref(table_a);
-  grpc_slice_hash_table_unref(table_b_larger);
-  grpc_slice_hash_table_unref(table_b_smaller);
-  grpc_slice_hash_table_unref(table_c);
-  grpc_slice_hash_table_unref(table_d);
-  grpc_slice_hash_table_unref(table_e);
-  grpc_slice_hash_table_unref(table_f);
-  grpc_slice_hash_table_unref(table_g);
-  grpc_slice_hash_table_unref(table_h);
+  GPR_ASSERT(TestHashTable::Cmp(*table_g, *table_h) != 0);
 }
 
+}  // namespace
+}  // namespace grpc_core
+
 int main(int argc, char** argv) {
   grpc_test_init(argc, argv);
   grpc_core::ExecCtx::GlobalInit();
-  test_slice_hash_table();
-  test_slice_hash_table_eq();
-  test_slice_hash_table_not_eq();
+  grpc_core::test_slice_hash_table();
```

Comment:
It'd be nice if this file could be made into a gtest one.

Reply:
Done.